qt-models/divelocationmodel.cpp

```diff
@@ -183,6 +169,56 @@ void LocationInformationModel::diveSiteDeleted(struct dive_site *, int idx)
 	endRemoveRows();
 }
 
+bool DiveSiteSortedModel::filterAcceptsRow(int sourceRow, const QModelIndex &source_parent) const
+{
+	// TODO: filtering
+	return true;
+}
+
+bool DiveSiteSortedModel::lessThan(const QModelIndex &i1, const QModelIndex &i2) const
+{
+	// The source indices correspond to indices in the global dive site table.
+	// Let's access them directly without going via the source model.
+	// Kind of dirty, but less effort.
+	struct dive_site *ds1 = get_dive_site(i1.row(), &dive_site_table);
+	struct dive_site *ds2 = get_dive_site(i2.row(), &dive_site_table);
+	if (!ds1 || !ds2) // Invalid dive sites compare as different
+		return false;
+	switch (i1.column()) {
+	case LocationInformationModel::NAME:
+	default:
+		return QString::localeAwareCompare(QString(ds1->name), QString(ds2->name)) < 0; // TODO: avoid copy
+	case LocationInformationModel::DESCRIPTION: {
```

Comment:
sorting by description seems... rather academic?

Reply:
Sure, but since the column header is there it should be clickable for consistency reasons, no? Consider it as an example of how to implement this, should someone like to add more columns.